### PR TITLE
Implement basic ignore user API

### PR DESCRIPTION
**Pull request by erikjohnston**

### CODE REVIEW

**Comment by NegativeMjark** [comment]
LGTM

File: synapse/storage/account_data.py [hunk 1]
```diff
@@ -60,6 +63,28 @@ def get_account_data_for_user_txn(txn):
             "get_account_data_for_user", get_account_data_for_user_txn
         )
 
+    @cachedInlineCallbacks(num_args=2)
+    def get_global_account_data_by_type_for_user(self, user_id, data_type):
```
Comment (NegativeMjark) [inline synapse/storage/account_data.py]: Given how this is used maybe this wants to be a cached list?

**Comment by NegativeMjark** [comment]
Would it be an idea to create a special cached database API for getting the ignore list that could populate the cache with a default empty list if the user hasn't set an ignore list?

**Comment by erikjohnston** [comment]
> Would it be an idea to create a special cached database API for getting the ignore list that could populate the cache with a default empty list if the user hasn't set an ignore list?

In terms of performance or code neatness?

**Pull request merged**